src/Raven.Server/Documents/Document.cs

```diff
@@ -13,6 +14,108 @@ public class TimeSeriesStream
         public IEnumerable<DynamicJsonValue> TimeSeries;
         public string Key;
     }
+
+    public sealed class QueriedDocument : Document, IResetSupport<List<QueriedDocument>>
```

Comment:
Let's move it to a separate file located somewhere in the area of query handling code.

Reply:
Sure, I left it here for development purposes only.